etcdmain/etcd.go

```diff
@@ -247,6 +248,11 @@ func startEtcd(cfg *config) (<-chan struct{}, error) {
 			l = netutil.LimitListener(l, int(fdLimit-reservedInternalFDNum))
 		}
 
+		l, err = transport.NewKeepAliveListener(l, u.Scheme, cfg.clientTLSInfo)
```

Comment:
maybe a comment here about avoiding wrapping a tls listener with a `LimitListener`? alternatively, it might be helpful to add a type assertion in `LimitListener` (or any other listener wrapper in the system) so it'll complain if anyone ever tries to wrap it around a tlslistener